lightning/src/ln/peer_handler.rs

```diff
@@ -1088,6 +1094,43 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 			}
 		};
 	}
+
+	/// This function can be called at an interval greater than every 30 seconds, it will:
+	/// disconnect the Peer if awaiting_pong is true
+	/// prepare to Ping all the Peers that the PeerManager is responsible for and set awaiting_pong to true
+	///
+	/// After calling this function one would want to call process_events() in order to finish Pinging the Peer
+	/// then assuming a pong Message is sent by the corresponding Peer one would 
+	/// call do_read_event() in order to reset awaiting_pong 
+	pub fn timer_tick_occured(&self) {
+		let mut peers_lock = self.peers.lock().unwrap();
+		let peers = peers_lock.borrow_parts();
+
+		for (descriptor, peer) in peers.peers.iter_mut() {
```

Comment:
Can this not be moved into the peers.retain() body, below? I think you should be able to put the entire function into there, but maybe not the last bit, I havent tried.

Reply:
So I originally had the functionality inside the loop on 1109 (i am assuming thats what your talking about) in the peers.retain() body, the compiler threw an error that said something along the lines of "the retain method requires unique access to *peer.peer so peers.peers_needing_send(descriptor) can't work".

I went ahead and moved the entire disconnect logic to its own loop on 1109 (I am assuming the other loop logic would not have worked as well)